internal/querynodev2/segments/manager.go

```diff
@@ -64,6 +65,25 @@ func WithID(id int64) SegmentFilter {
 	}
 }
 
+type SegmentAction func(segment Segment) bool
+
+func IncreaseVersion(version int64) SegmentAction {
+	return func(segment Segment) bool {
+		log := log.Ctx(context.Background()).With(
+			zap.Int64("segmentID", segment.ID()),
+			zap.String("type", segment.Type().String()),
+			zap.Int64("segmentVersion", segment.Version()),
+			zap.Int64("updateVersion", version),
+		)
+		if segment.Version() >= version {
+			log.Warn("segment version cannot go backwards, skip update")
+			return false
+		}
+		segment.UpdateVersion(version)
```

Comment:
use CAS to protect concurrent writes here as `UpdateSegmentBy` acquires a read lock only